src/tools/idset.jl

```diff
@@ -3,18 +3,16 @@ struct IdSet{T} <: AbstractSet{T}
   IdSet{T}() where T = new(IdDict{T,Nothing}())
 end
 
-Base.eltype(::IdSet{T}) where T = T
+IdSet(xs) = IdSet{eltype(xs)}(xs)
```

Comment:
Is this needed anymore?

Reply:
Nevermind, its deleted and redefined in this PR.

Reply:
I think so, unless we remove the T from `IdSet{T}(xs)` and deduce it internally